.github/workflows/publish-containers.yml

```
@@ -24,68 +24,17 @@ jobs:
           IMAGE_NAME=$(echo ${{ github.repository }} | tr '[:upper:]' '[:lower:]')
           echo "image_name=$IMAGE_NAME" >> $GITHUB_OUTPUT
 
-      # Outputs container tags for tagged pushes starting by 'v'
-      # Pushes only to GitHub Container Repository (ghcr.io)
-      #
-      # Tags stable versions as :latest
-      # Pre-releases, alphas, etc. as :snapshot
-      - name: Prepare Image Tags from Git Tag
-        id: tagged
-        if: ${{ startsWith(github.ref, 'refs/tags/') }}
-        run: |
-          set -x
-
-          # Remove everything else than the tagged version
-          VERSION=${GITHUB_REF#refs/tags/}
-
-          # Tag as stable (:latest) if there is no letters in the version
-          # Otherwise, tag as preview (:snapshot)
-          if [[ $VERSION =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
-            # Matches 1.0.0, 1.0.1, etc.
-            NAMED_VERSION="latest"
-          else
-            # Matches 1.0.0a1, 1.0.0rc1, etc.
-            NAMED_VERSION="snapshot"
-          fi
-
-          TAGS=$"\
-          ghcr.io/${{ steps.image.outputs.image_name }}:${NAMED_VERSION},\
-          ghcr.io/${{ steps.image.outputs.image_name }}:${VERSION}\
-          "
-
-          # Output the target tags
-          echo "
-          CONTAINER_TAGS=${TAGS}
-          NAMED_VERSION=${NAMED_VERSION}
-          VERSION=${VERSION}
-          " >> "${GITHUB_ENV}"
-
-      # Outputs the containers tags for staging branches
-      # Only publishes them onto Docker Hub
-      - name: Prepare Tags for Staging
-        id: staging
-        if: ${{ startsWith(github.ref, 'refs/heads/') }}
-        run: |
-          set -x
-
-          # Version name is the branch name
-          # Slashes are substitued by dashes
-          CLEAN_BRANCH_NAME=$(echo ${GITHUB_REF#refs/heads/} | tr / -)
-
-          # Target GHCR with the branch name as the tag, e.g. unstable-main
-          TAGS=$"\
-          ghcr.io/${{ steps.image.outputs.image_name }}:unstable-${CLEAN_BRANCH_NAME}
-          "
-
-          # Set version name for open-containers version label as:
-          # <branch>-<drift>-g<6-digit-hash>
-          NAMED_VERSION=$(git describe --all --long | tr -d $'\n')
-
-          # Output the target tags
-          echo "
-          CONTAINER_TAGS=${TAGS}
-          NAMED_VERSION=${NAMED_VERSION}
-          " >> "${GITHUB_ENV}"
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: |
+            ${{ steps.image.outputs.image_name }}
+          tags: |
+            type=ref,event=branch,prefix=unstable-
+            type=pep440,pattern={{version}}
+            type=pep440,pattern={{major}}.{{minor}}
```

Shouldn't it be like so?
```suggestion
            type=pep440,event=tag,pattern={{version}}
            type=pep440,event=tag,pattern={{major}}.{{minor}}
```

---

`event=tag` is a parameter for `type=ref`, it generates a label when a new reference type is added (tag, head, etc.). `type=pep440` only works on tags.